src/components/DocumentEditPage.js

```diff
@@ -0,0 +1,116 @@
+import Editor from "./Editor.js";
+import { setItem, getItem, removeItem } from "../utils/storage.js";
+import { request } from "../api/api.js";
+import { push } from "../utils/router.js";
+
+export default function DocumentEditPage({ $target, initialState }) {
+  const $editPage = document.createElement("div");
+  $editPage.className = "edit-page";
+  this.state = initialState;
+
+  let documentLocalSaveKey = `temp-document-${this.state.documentId}`;
+
+  const tempDocument = getItem(documentLocalSaveKey, {
+    title: "",
+    content: "",
+  });
+
+  let timer = null;
+
+  const editor = new Editor({
+    $target: $editPage,
+    initialState: tempDocument || {
+      title: "",
+      content: "",
+    },
+    onEditing: (document) => {
+      if (timer !== null) {
+        clearTimeout(timer);
+      }
+
+      timer = setTimeout(async () => {
+        setItem(documentLocalSaveKey, {
+          ...document,
+          tempSaveData: new Date(),
+        });
+
+        await request(`/${this.state.documentId}`, {
+          method: "PUT",
+          body: JSON.stringify(document),
+        });
+
+        if (this.state.document.title) {
+          if (this.state.document.title !== document.title) {
+            push({
+              type: null,
+              id: null,
+            });
+          }
+        }
+        removeItem(documentLocalSaveKey);
+      }, 500);
+    },
+  });
+
+  this.setState = async (nextState) => {
+    if (this.state.documentId !== nextState.documentId) {
+      documentLocalSaveKey = `temp-document-${this.state.documentId}`;
+      this.state = nextState;
+
+      if (this.state.documentId === "new") {
+        editor.setState(
+          this.state.document || {
+            title: "",
+            content: "",
+          }
+        );
+      } else {
+        await fetchDocument();
+      }
```

Comment:
p4: 빈 게시글로 상태변경하는 로직이 반복되는 것 같은데 이런식으로 할수도 있을 것 같아요?
```suggestion
   function getEmptyDocument(){
      return {title:'',content:''}
   }

      if (this.state.documentId === "new") {
        editor.setState(
          this.state.document || getEmptyDocument()
        );
      } else {
        await fetchDocument();
      }
```